editor.planx.uk/src/lib/lowcalStorage.ts

```diff
@@ -145,7 +146,9 @@ const getPublicContext = (sessionId: string) => ({
   context: {
     headers: {
       "x-hasura-lowcal-session-id": sessionId,
-      "x-hasura-lowcal-email": useStore.getState().saveToEmail?.toLowerCase(),
+      "x-hasura-lowcal-email":
```

Comment:
Email address cannot and should not be unique in this table as it's perfectly valid for one email address to be associated with multiple sessions.

If we allow `""` as an email address, the `public` role can perform `SELECT` and `UPDATE` operations on any record where the email is `""` if they know the id, via our GraphQL API endpoint. The permissions for this are defined in `tables.yaml`

Example from Pizza using request headers - 

<img width="1433" alt="Screenshot 2023-02-24 at 18 20 34" src="https://user-images.githubusercontent.com/20502206/221258910-2302544c-e15f-4b38-a408-21f04da2069b.png">

I know we did speak about this a while back sorry, and I don't remember the conclusion we came to exactly - I might be treading over ground we've already covered. Is there a current use case for having a "Pay" component in a flow without a "Send" component? 

If there is, could we solve this by also checking for "Pay" components in `isSaveReturnFlow()` as opposed to loosening this restriction? This would mean we'd always have an email address to associate with a session, but there are implications in how we'd handle a session being completed as we'd need to sub the `submitted_at` logic with a `completed_at` / `paid_at` or similar.

Reply:
Actually, another (simpler and more flexible?) option here is to update the permissions to allow `SELECT` and `UPDATE` if... 
 - You know the session_id
 - You know the email address
 - The email address is not `""`

This would allow a non Save and Return flow to allow payments, keep sessions and audit logs, and not risk exposing session data.

These rules are set in the "Permission" tab of a table in Hasura (Data > lowcal_sessions > Permission > Public > Update / Select > Column / Row update rules).

If the above makes sense, I'd suggest going for this and trying it out 👍